src/plugins/usage_collection/public/plugin.ts

```diff
@@ -90,7 +95,7 @@ export class UsageCollectionPlugin implements Plugin<UsageCollectionSetup> {
 
   public start({ http, application }: CoreStart) {
     if (!this.reporter) {
-      return;
```

Comment:
Not sure the best way to handle this but it _shouldn't_ ever happen as far as I understand it?

Reply:
Yes, i believe its there since `private reporter?: Reporter;` is not initialized by the constructor so it must be optional.

Reply:
Right. Now we have to always return the contract, so I switched it to throw in this ("impossible"?) case.

Reply:
Throwing an error sounds reasonable here, if for any  reason it was not registered consumers will start throwing -harder to understand- errors anyways.